frontend/src/app/shared/utils/option.ts

```diff
@@ -9,7 +9,17 @@ export class Option {
   static of<OptionValue extends PrimitiveOption>(value: OptionValue, displayValue?: string): SelectOption<OptionValue> {
```

Comment:
Well, it actually depends how precise we want the type information to be.
Consider the following set of data:
```
const bool = true;
const string = 'test';
```

With generics we would have the following types:
```
 Option.of(bool);  // SelectOption<true>
 Option.of(string);  // SelectOption<'test'>
```
without generics (using PrimitiveOption) we would have:
```
 Option.of(bool);  // SelectOption<PrimitiveOption>
 Option.of(string);  // SelectOption<PrimitiveOption>
```

So it depends how strict type information we want to keep. This will be even more prevalent for Options.fromMap (see the respective comment)

Reply:
Yeah, you're right. OK, let's got with full generics!